external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala

```diff
@@ -317,6 +353,8 @@ private[kafka010] case class KafkaSource(
               try {
                 result = Some(body)
               } catch {
+                case x: OffsetOutOfRangeException =>
```

Comment:
OffsetOutOfRange exeception can happen on any poll.  I agree that with the way things are now, it shouldn't happen, but that has more to do with us currently having an explictly enforced value for auto.offset.reset, not with whether data is being fetched. This change is basically paranoia about catching all nonfatal errors without respecting user settings regarding data loss.